### PR TITLE
Add optional RMT testing to agama integration tests

### PR DESCRIPTION
**Pull request by rtamalin**
Update the build/ci/run-agama-rust-tests script to add a section at the end that conditionally runs the `rmt` test tool, if `RMT_HOST` is non-empty, to exercise RMT registrations with support for downloading the RMT's self-signed cert if the `RMT_HOST` url is specified using the `https` protocol. Also corrected the text of the activation group entry, and added the `rmt` tool to the set of tools being built.

Update README.md to add a section to testing about running the Agama rust integration tests. Also updated the `.env` configuration sections, wrapped some long lines, added references to the testing sections in the earlier examples of running those tests, and removed some trailing white space on lines throughout the file.

Update the .env-example file to include a commented out `RMT_HOST` line.

Update the GitHub Actions Agama workflow to setup RMT_HOST (based upon a secret of the same name) in the environment used by the test jobs.

Depends On: agama-project/agama#3640

Implements: SCC-805

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -188,9 +240,9 @@ The available workflows, and their associated trigger events can be listed by
 running the `act --list` or `gh act --list` command, as follows:
 
 ```bash
-% act --list   
-INFO[0000] Using docker host 'unix:///var/run/docker.sock', and daemon socket 'unix:///var/run/docker.sock' 
-Stage  Job ID                Job name                                        Workflow name           Workflow file  Events      
+% act --list
+INFO[0000] Using docker host 'unix:///var/run/docker.sock', and daemon socket 'unix:///var/run/docker.sock'
```
Comment (felixsch) [inline README.md]: I think those info lines could be removed or? While they are in the actual output, they do not help in the examples here or? Same applies to the other `INFO[0000]` lines.

File: build/ci/run-agama-rust-tests [hunk 2]
```diff
@@ -89,14 +93,40 @@ export RUSTFLAGS="-L ${STDCXX_DIR}"
...
 group "deregister the system and cleanup registration state"
 ${CONNECT_OUT}/suseconnect -d
 ${CONNECT_OUT}/suseconnect --clean
 groupend
+
+# optionally run RMT verification tests if RMT_HOST is non-empty
+if [[ -n "${RMT_HOST}" ]]; then
+	rmt_cert=
+
+	if [[ ${RMT_HOST} =~ https://.* ]]; then
```
Comment (felixsch) [inline build/ci/run-agama-rust-tests:112]: How about adding a debug output here to show from which URL the `crt` is fetched to make debugging easier

**Review by felixsch — APPROVED** [review]
lgtm 👍 

```
::groupend::
::group::run the activation example
Ok(Credentials { login: "SCC_a4a1a650577c482ba4b5a47d5515c592", password: "67426cebdd4a48d3" })
Ok(())
Ok(Service { name: "SUSE_Linux_Enterprise_Server_16.0_x86_64", url: "https://scc.suse.com/access/services/2714?credentials=SUSE_Linux_Enterprise_Server_16.0_x86_64" })
::groupend::
::group::deregister the system and cleanup registration state
Deregistering system to SUSE Customer Center
-> Removing service from system ...

Cleaning up ...
Successfully deregistered system
::groupend::
::group::retrieve rmt cert
::groupend::
::group::run the rmt example
Certificate path provided: /tmp/rmt_host.crt. Processing...
Copied 1265 bytes to /etc/pki/trust/anchors/rmt_host.crt
Running update-ca-certificates...
CA certificates updated successfully.
Ok(Credentials { login: "SCC_cdc37b3a3f4b499ea6594284d602c853", password: "902f3214e904419d" })
Ok(())
Ok(Service { name: "SUSE_Linux_Enterprise_Server_x86_64", url: "https://rmt.scc.suse.de/services/2930?credentials=SUSE_Linux_Enterprise_Server_x86_64" })
::groupend::
::group::deregister the system from RMT and cleanup registration state
Deregistering system to registration proxy https://rmt.scc.suse.de
-> Removing service from system ...

Cleaning up ...
Successfully deregistered system
::groupend::
```

**How I reviewed this merge request:**

- Checked the workflow setup and suggested my nitpicks
- Run the Agama test against our local RMT to see it working in action

**As always, if you think I missed something I should test, please let me know!**:rocket:

cheers,

File: build/ci/run-agama-rust-tests (hunk 2 above)
Comment (rtamalin) [reply build/ci/run-agama-rust-tests:112]: Sure ;-)

File: README.md (hunk 1 above)
Comment (rtamalin) [reply README.md]: They are examples of the output from running the command... Happy to trim then if we don't feel they are beneficial ;-)

Comment (felixsch) [reply README.md]: I would remove them since they are system specific, no?

File: build/ci/run-agama-rust-tests (hunk 2 above)
Comment (rtamalin) [reply build/ci/run-agama-rust-tests:112]: Resolved

File: README.md (hunk 1 above)
Comment (rtamalin) [reply README.md]: Resolved

**Comment by felixsch** [comment]
nice still lgtm!
